### DISCUSSION TITLE
View Collection SQL Query

### DISCUSSION THREAD

**Discussion by ivanpajon** [pocketbase/pocketbase#3873] [Q&A] [answered]
I have two Auth Collections with same schema: **coaches** and **clients**. And I want to create a View Collection of the usernames used in both tables to validate each username is not repeated in the other table.

Now, I get Pocketbase to work if the Select Query of my View Collection is `SELECT id, username FROM (SELECT id, username FROM coaches UNION SELECT id, username FROM clients)` but if I only do `SELECT id, username FROM coaches UNION SELECT id, username FROM clients)` it gives me an "Invalid query - invalid identifier part - expected "as", got UNION." error.
![view_collection_error](https://github.com/pocketbase/pocketbase/assets/17519714/6b893793-f284-419a-b069-9d453579b0e7)

I don't know if I'm doing something wrong or if there is other way to accomplish what I trying to do.
Thank you in advance.

**Comment by ganigeorgiev** [comment, marked as answer]
Root level `UNION` selects are not supported (see https://github.com/pocketbase/pocketbase/discussions/1938).

An alternative, could be to wrap your query in a subselect, aka. `SELECT id, username FROM ( ... )`

**Reply by ganigeorgiev** [reply]
_Side-note about "I have two Auth Collections with same schema..." - if all fields are the same you may find it easier to keep only 1 auth collection with a `select` "type" field or similar to differentiate between "clients" and "coaches"_

**Reply by ivanpajon** [reply]
Initially I only had one collection Users, but Coaches and Clients rely on different relationships and some of them was not possible with only 1 table. Thanks for the suggestion.

If I make a nested query like `SELECT id, username ( ... )` will not impact the performance? (I ask from ignorance)

**Reply by ganigeorgiev** [reply]
> If I make a nested query like `SELECT id, username ( ... )` will not impact the performance?

No, there shouldn't be any significant difference. The only drawback of this I think is that the `username` will be treated as `json` field type instead of `text` since it will be an expression and because of some internal optimizations if you have a numeric username value it may get serialized as number. To force always a string value you can try to cast it, something like:
```sql
SELECT id, CAST(username as text) as username
FROM (
  SELECT id, username FROM coaches
  UNION SELECT id, username FROM clients
)
```

**Reply by ivanpajon** [reply]
You were right.
Before (without CAST):
![before](https://github.com/pocketbase/pocketbase/assets/17519714/c2f7ca4e-88b3-4363-ac28-0811ef19da9c)

After (with CAST):
![after](https://github.com/pocketbase/pocketbase/assets/17519714/04033407-75af-4ca8-819d-95e6b4f77f55)

Thanks a lot!

**Reply by humbertomorais** [reply]
@ganigeorgiev Pocketbase is increasingly surprising and always bringing more functionality. May the project continue to grow and bring more features. I am very grateful to be able to use the Pocketbase SaaS service.
